.github/workflows/codeql.yaml

```diff
@@ -4,6 +4,9 @@ on:
   push:
     branches:
     - main
+  pull_request:
+    branches:
+    - main
```

Comment:
It only takes ~ 8m to complete (ref: https://github.com/kubernetes-sigs/secrets-store-csi-driver/actions/runs/3331865967), so enabling it as part of pull requests